### PR TITLE
Revisions: Revision display/browsing

### PR DESCRIPTION
**Pull request by nheagy**
Refs #10304 

This implements the basic revision viewing & browsing.

revisions display mk2

To test:
 - Edit a post with revision history
 - Tap `…` then `History`
 - Tap a revision
 - Ensure the UI behaves as expected

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @IBOutlet var previousButton: UIButton!
+    @IBOutlet var nextButton: UIButton!
+
+
+    private lazy var doneBarButtonItem: UIBarButtonItem = {
+        let doneItem = UIBarButtonItem(barButtonSystemItem: .done, target: nil, action: nil)
+        doneItem.on() { [weak self] _ in
+            self?.dismiss(animated: true)
+        }
+        doneItem.title = "Done"
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift]: Should this be localized?

File: WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+}
+
+class RevisionDiffsBrowserViewController: UIViewController {
+    var revisionState: RevisionBrowserState?
+    var diffVC: RevisionDiffViewController?
+    var operationVC: RevisionOperationViewController?
+    @IBOutlet var revisionTitle: UILabel!
+    @IBOutlet var previousButton: UIButton!
+    @IBOutlet var nextButton: UIButton!
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift]: Can we make these private?

File: WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    func currentRevision() -> Revision {
+        return revisions[currentIndex]
+    }
+    mutating func decreaseIndex() {
+        currentIndex = max(currentIndex - 1, 0)
+    }
+    mutating func increaseIndex() {
+        currentIndex = min(currentIndex + 1, revisions.count)
+    }
```
[the remark this answers is not in the document]
Comment (danielebogo) [reply WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift]: I love injecting a container too, with the purpose to populate the data!

**Comment by iamthomasbishop** [comment]
@nheagy looking awesome - nice work! I have a couple questions/notes:

* Can we make the chevrons (left/right arrows on the revision detail) `darkGrey` by default? Additionally, I see we're removing the icons altogether when the user is viewing the first or last version in the list. I was originally intending on showing "disabled" chevrons, in `greyLighten20`, but if you think we should just hide them altogether when disabled, that's fine.

* I also noted on another PR from @danielebogo's, but the colors of the `+/-` icons are reversed, so I think that'll be addressed in that other PR

* Curious – how does the canvas/post preview look compared to the same thing in Aztec? 

* Is the `Load` button/action on the Revision Detail being added in a separate PR?

File: WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift]: Ya, I think you're right and it should be a class. I wanted to use a struct but then wrote the index modifiers this way. 👍

File: WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift (hunk 1 above)
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Post/Revisions/Views/RevisionDiffsBrowserViewController.swift]: Oh darn, ya, that was meant to be temporary 😬

**Comment by nheagy** [comment]
> Can we make the chevrons (left/right arrows on the revision detail) darkGrey by default

Done 👍 

> Curious – how does the canvas/post preview look compared to the same thing in Aztec?

It's different. I'll probably leave tweaking that for another PR, unless I can find a solution quickly.

> Is the Load button/action on the Revision Detail being added in a separate PR?

Yup 😀

**Comment by danielebogo** [comment]
Awesome @nheagy ! Thanks a lot!

**Comment by iamthomasbishop** [comment]
@nheagy would you be able to drop an updated screenshot of the changes when you get a chance so I can give it a final design pass?

**Comment by nheagy** [comment]
@iamthomasbishop done! Let me know any revisions, and I'll put them into new, high-priority PRs.

**Comment by iamthomasbishop** [comment]
Awesome, thanks! :shipit:

**Pull request merged**